crates/adapters/src/test/mock_dezset.rs

```diff
@@ -318,6 +336,102 @@ where
     }
 }
 
+#[derive(Clone)]
+pub struct MockZSetArrowStream<T, U, C>
+where
+    T: Send + Sync,
+    U: Send + Sync,
+{
+    buffer: MockDeZSetStreamBuffer<T, U>,
+    config: C,
+}
+
+impl<T, U, C> MockZSetArrowStream<T, U, C>
+where
+    T: Send + Sync,
+    U: Send + Sync,
+    C: Clone,
+{
+    pub fn new(handle: MockDeZSet<T, U>, config: C) -> Self {
+        Self {
+            buffer: MockDeZSetStreamBuffer::new(handle),
+            config,
+        }
+    }
+}
+
+impl<T, U, C> InputBuffer for MockZSetArrowStream<T, U, C>
+where
+    T: for<'de> DeserializeWithContext<'de, SqlSerdeConfig> + Hash + Send + Sync + 'static,
+    U: for<'de> DeserializeWithContext<'de, SqlSerdeConfig> + Hash + Send + Sync + 'static,
+    C: Clone + Send + Sync + 'static,
+{
+    fn flush(&mut self) {
+        self.buffer.flush()
+    }
+
+    fn take_some(&mut self, n: usize) -> Option<Box<dyn InputBuffer>> {
+        self.buffer.take_some(n)
+    }
+
+    fn hash(&self, hasher: &mut dyn Hasher) {
+        self.buffer.hash(hasher)
+    }
+
+    fn len(&self) -> usize {
+        self.buffer.len()
+    }
+}
+
+impl<T, U> ArrowStream for MockZSetArrowStream<T, U, SqlSerdeConfig>
+where
+    T: for<'de> DeserializeWithContext<'de, SqlSerdeConfig>
+        + Hash
+        + Send
+        + Sync
+        + Debug
+        + Clone
+        + 'static,
+    U: for<'de> DeserializeWithContext<'de, SqlSerdeConfig>
+        + Hash
+        + Send
+        + Sync
+        + Debug
+        + Clone
+        + 'static,
+{
+    fn insert(&mut self, data: &RecordBatch) -> AnyResult<()> {
+        let deserializer = ArrowDeserializer::from_record_batch(data)?;
+        let deserializer =
+            &mut <dyn ErasedDeserializer>::erase(deserializer) as &mut dyn ErasedDeserializer;
+
+        let records = Vec::<T>::deserialize_with_context(deserializer, &self.config)?;
+        self.buffer.updates.extend(
+            records
+                .into_iter()
+                .map(|r| MockUpdate::<T, U>::with_polarity(r, true)),
+        );
+
+        Ok(())
+    }
+
+    fn delete(&mut self, _data: &RecordBatch) -> AnyResult<()> {
+        todo!()
```

Comment:
will these be done in this PR?

Reply:
It's an unused part of the test harness.